# Not allow parse date string (`ts-compat/no-date-parse`)

ğŸ’¼ This rule is enabled in the âœ… `recommended` config.

<!-- end auto-generated rule header -->

There is diffrent date string format support is diffrent browsers. 'yyyy/MM/dd' is the common date format. However, backend or local requestor give the format may be 'yyyy-MM-dd'. It is recommended use lib to parse.

## Rule Details

Examples of **incorrect** code for this rule:

```ts
var str = "2010/03/01 00:00:00";
new Date(str);
Date.parse(str);
```

Examples of **correct** code for this rule:

```ts
new Date(2021, 0, 1);
new Date(1700000000000);
```

## Options

### allowAny

Boolean. When true allowed any.

## When Not To Use It

Give date string that format is 'yyyy/MM/dd'.

## Further Reading

[ç”±äºæµè§ˆå™¨å·®å¼‚å’Œä¸ä¸€è‡´æ€§ï¼Œå¼ºçƒˆå»ºè®®ä¸è¦ä½¿ç”¨Dateæ„é€ å‡½æ•°ï¼ˆå’ŒDate.parseï¼Œå®ƒä»¬æ˜¯ç­‰æ•ˆçš„ï¼‰è§£ææ—¥æœŸå­—ç¬¦ä¸²ã€‚](https://developer.mozilla.org/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Date#%E7%A4%BA%E4%BE%8B)
