# Prefer immuable url (`ts-compat/prefer-immuable-url`)

ğŸ’¼ This rule is enabled in the âœ… `recommended` config.

<!-- end auto-generated rule header -->

Most URL Polyfills do not support url setter. Aviod setting property, make more compat.

## Rule Details

Examples of **incorrect** code for this rule:

```ts
var url = new URL(location.href);
url.hash = "#foo";
```

Examples of **correct** code for this rule:

```ts
var url = new URL(location.href);
url = new URL("#foo", url);
```

## When Not To Use It

Use URL Polyfill what support setter.

## Further Reading

[sky-coreæ˜¯ä¸€ä¸ªå¤„ç†æµè§ˆå™¨å…¼å®¹é—®é¢˜çš„polyfillåº“å’Œå·¥å…·åº“](https://www.npmjs.com/package/sky-core)
